akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -219,6 +251,7 @@ private[remote] abstract class InboundCompression[T >: Null](
   /** Drop `activeTable` and start using the `nextTable` in its place. */
   private def startUsingNextTable(): Unit = {
     log.debug("Swaping active decompression table to version {}.", nextTable.version)
+    oldTable = activeTable
     activeTable = nextTable
     nextTable = DecompressionTable.empty
     // TODO we want to keep the currentTableVersion in State too, update here as well then
```

Comment:
the swapping should likely be done in an atomic op, to avoid weird situations.
Relates to the race I mentioned in Decoder.
Solving it would perhaps be done well via making the Compressions extends AtomicRef(state), but likely expensive.